### PR TITLE
Removed export/view permission from a dashboard viewer

### LINKED ISSUE #2309: Viewer users can export/view Dashboards?
Hi. I'm working as a Viewer user and I can export/view dashboard json. 

image

IMHO Viewer users shouldn't be able to view metric names/paths/functions if not explicitly explained somewhere by the dashboard editor user

Export and View JSON is a useful tool for editor users but not for viewer users.

### PR DESCRIPTION
**Pull request by utkarshcmu**
Fixed #2309

### CODE REVIEW

**Comment by torkelo** [comment]
any reason for this?

**Comment by utkarshcmu** [comment]
Just read the comments in #2309, saw you agreeing with @toni-moreno and I also thought the same way that these options might not be useful to the viewers. So, implemented this small change.

But sometimes, the dashboard creators do not want the viewers/users to see what are the metric names, tags, metric query, etc. They just want viewers to see the graphs and thats it. But I agree that export/view options doesn't hurt any dashboard from being edited even if they are available. So, we can keep these options for the viewers if you think and we can close this PR. :) 

But its great working with you on this project. Lets make Grafana the best monitoring dashboard solution. :)

**Comment by toni-moreno** [comment]
Hi @torkelo. This PR is exactly the expected behaviour for me.

I know advanced users could see this information by only open the browser developer tools. But we prefer don't offer explicitly

**Comment by torkelo** [comment]
@toni-moreno I think I agree, just wanted to know the reasoning behind the PR. If it's only for aesthetics reasons I think it could be a good idea (the cog menu and actions are very rarely used by viewers and only in the way, don't make sense etc). But if its for security reasons it's just stupid :)

**Comment by utkarshcmu** [comment]
@torkelo Thanks for explaining the network traffic logic and chrome dev tools thing. I was not aware of it. It does not make sense now to hide these options from Viewers just for the security. I agree. :)

So, about hiding the options from viewers as they are not mostly useful to them, we can keep the default behavior of being these options visible as they are now. 

How about having options in the "Settings" view to hide the controls from the viewers like shown below?

screen shot 2015-10-27 at 10 39 16 am

Then the editors/admins can control/configure what Viewers should see and what they should not. :)

**Comment by utkarshcmu** [comment]
@torkelo Any decisions on this?

File: public/app/features/dashboard/partials/dashboardTopNav.html [hunk 1]
```diff
@@ -29,14 +29,14 @@
 				<li ng-show="dashboardMeta.canSave">
 					<a ng-click="saveDashboard()" bs-tooltip="'Save dashboard'" data-placement="bottom"><i class="fa fa-save"></i></a>
 				</li>
-				<li class="dropdown">
+				<li ng-if="dashboardMeta.canEdit" class="dropdown">
```
Comment (bergquist) [inline public/app/features/dashboard/partials/dashboardTopNav.html]: This makes it impossible to make a dashboard editable again if its none editable (Settings -> Toggles -> Editable ) even for admins.

Comment (bergquist) [inline public/app/features/dashboard/partials/dashboardTopNav.html]: ng-if-"dashboardMeta.canEdit || contextSrv.isEditor"
Would solve that. But I really dislike conditions like that in the views.

**Comment by utkarshcmu** [comment]
@bergquist I understand having these "if" conditions in the views is not ideal but I think this these "If" statements are at a single place, so easy to handle in the future as well. 

This PR should now not allow the Viewers to have the "Manage dashboard" cog icon. :)

**Comment by bergquist** [comment]
Making the UI easier for any user will always have our attention.

I'm going merge this and make some changes. (code trumps talking about code :))
Feel free to reopen PR if you don't like my changes

**Comment by Turum** [comment]
I see that in a third version Export and Json panels are available for viewers again. Is it possible to hide it? Sorry for molestation in this matter.
Thanks.

**Comment by torkelo** [comment]
@Turum no, why? It's not like they can't see the dashboard json anyway (looking at http traffic), the in browser frontend app need the dashboard json to show the dashboard.

**Comment by Turum** [comment]
I agree if we are talking about tech guys, if they really want they can trace network traffic, but if we make dashboards for business user we need to restrict access to almost everything that could cause to additional questions and I really doubt that they will engaged in tracing traffic.

**Comment by geraybos** [comment]
I found export/view permission from a dashboard viewer has been enabled again in v5.0.2. 

This function was added again?

**Comment by JoeeGrigg** [comment]
Is there any update on this?

I would agree with @Turum. We should be able to hide things from the UI that could possibly confuse users. It isn't a security feature at all but a UX feature. I know for a fact that if one of the users that I support found the Share -> Export section they would get confused and probably start asking questions.

It just seems like something that should be available as an option to disable.

**Pull request merged**